### PR TITLE
ref(ui): Use flexbox for modal footer

### PR DESCRIPTION
**Pull request by evanpurkhiser**
😬 🙏 percy

I hand checked a few modals, looks to be good.

This fixes an alignment issues with buttons with and without icons next to each other

### CODE REVIEW

File: src/sentry/static/sentry/less/shared-components.less [hunk 1]
```diff
@@ -2724,6 +2724,9 @@ ul.radio-inputs {
     padding: 20px 30px;
     border-top: 1px solid @trim;
     margin: 20px -30px -30px;
+    display: flex;
+    justify-content: flex-end;
+    text-align: initial;
```
Comment (Chrissy) [inline src/sentry/static/sentry/less/shared-components.less]: `text-align` is an inherited property and therefore likely to cause bugs. What exactly is this overriding?

**Review by Chrissy — APPROVED** [review]
Generally looks good with one warning ⚠️

File: src/sentry/static/sentry/less/shared-components.less (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/less/shared-components.less]: Because `text-align` gets set to `right` by the bootstrap class.

Comment (Chrissy) [reply src/sentry/static/sentry/less/shared-components.less]: 👍

**Pull request merged**